liquibase-standard/src/main/java/liquibase/database/AbstractJdbcDatabase.java

```diff
@@ -168,6 +167,16 @@ public boolean getAutoCommitMode() {
         return !supportsDDLInTransaction();
     }
 
+    /**
+     * Used for TEST only
+     *
+     * @param previousAutoCommit
+     *
+     */
+    public void setPreviousAutoCommit(Boolean previousAutoCommit) {
+        this.previousAutoCommit = previousAutoCommit;
+    }
```

Comment:
I am pretty sure that the spock tests (or possibly groovy) let you access private fields and set their value with only a warning.